Client/mods/deathmatch/logic/CClientModel.cpp

```diff
@@ -125,4 +136,33 @@ void CClientModel::RestoreEntitiesUsingThisModel()
 
     // Restore DFF/TXD
     g_pClientGame->GetManager()->GetDFFManager()->RestoreModel(m_iModelID);
+
+    return true;
+}
+
+bool CClientModel::AllocateTXD(std::string &strTxdName)
+{
+    uint uiSlotID = g_pGame->GetPools()->AllocateTextureDictonarySlot(m_iModelID - MAX_MODEL_DFF_ID, strTxdName);
+    if (uiSlotID == -1)
+        return false;
+
+    return true;
+}
+
+bool CClientModel::DeallocateTXD(CModelInfo* pModelInfo)
+{
+    uint uiTextureDictonarySlotID = pModelInfo->GetModel() - MAX_MODEL_DFF_ID;
+
+    for (uint uiModelID = 0; uiModelID < MAX_MODEL_DFF_ID; uiModelID++)
+    {
+        CModelInfo* pModelInfo = g_pGame->GetModelInfo(uiModelID, true);
```

Comment:
What if `pModelInfo` is `nullptr`?

Reply:
`pModelInfo` can't be `nullptr` here because second arg for `GetModelInfo` is `true`